### ISSUE TITLE
Parameter validation too strict

### ISSUE THREAD

**Issue by michaelom** [swagger-api/swagger-parser#1063]
#960 introduced a bug:

This appears to be a valid definition:
```yaml
  /customers/{id}/legal-person/addresses/{addressId}:
    parameters:
      - $ref: "#/components/parameters/id"
      - $ref: "#/components/parameters/addressId"

```

But results in the following `messages` after Parsing:

```
attribute paths.'/customers/{id}/legal-person/addresses/{addressId}'.There are duplicate parameter values
```

Code:
```
 val parseResult = OpenAPIV3Parser().readWithInfo("http://localhost:8080/swagger/openapi.yaml", emptyList())

            val errors = parseResult.messages.joinToString(separator = System.getProperty("line.separator"))
            print(errors)

```

**Comment by cvgaviao** [comment]
Hello @jakubmalek, I did validate your patch and confirmed that it fixed the validation messages that I was facing with multiple parameter references at #1066. 
But I'm wondering... 
Why I'm getting a parameter with null values (including its name) when I do:
```
PathItem path = api.getPaths().get("/anPath/{anParameter}");
Parameter anParam = path.getParameters().get(0);
assertThat(anParam.getName()).isEqualTo("anParameter");
```  
those operation parameters wouldn't be resolved at parsing stage ?

**Comment by gracekarina** [comment]
closed by #1065 and #1080
